Extending/FileSystemProviders/index.md

```diff
@@ -1,48 +1,159 @@
+By default, Umbraco uses an instance of `PhysicalFileSystem` to handle the storage location of the media archive (~/media).
 
-```xml
-<Provider alias="media" type="Umbraco.Core.IO.PhysicalFileSystem, Umbraco.Core">
-    <Parameters>
-        <add key="virtualRoot" value="~/media/" />
-    </Parameters>
-</Provider>
-```  
+This can be configured during composition:
+
+```csharp
+using Umbraco.Core.Composing;
+using Umbraco.Core;
+using Umbraco.Core.IO;
+
+namespace Umbraco8Examples.Composition
+{
+    public class SetMediaFileSystemComposer : IUserComposer
+    {
+        public void Compose(Umbraco.Core.Composing.Composition composition)
+        {
+            composition.SetMediaFileSystem(() => new PhysicalFileSystem("~/custommediafolder"));
+        }
+    }
+}
+```
+The PhysicalFileSystem accepts the application path to the 'virtual Root' to location the Media folder the ~ is important!
+
+Alternatively you can combine:
+a **RootPath** - the rooted, filesystem path, using directory separator chars, NOT ending with a separator // eg "c:" or "c:\path\to\site" or "\\server\path"
+and **RootUrl** - the relative url, using url separator chars, NOT ending with a separator // eg "" or "/Views" or "/Media" or "/<vpath>/Media" in case of a virtual path
+
+:::note
+In Umbraco V7, this configuration was located a physical file: `/config/FileSystemProviders.config`.
+:::
 
 `PhysicalFileSystem` implements the `IFileSystem` interface, and it is possible to replace it with a custom class - eg. if you want your media files stored on Azure or something similar.
 
 If you configure Umbraco to use a custom file system provider for media, you most likely won't need to access the implementation directly. Umbraco uses a wrapper class called `MediaFileSystem`. You can get a reference to this wrapper class with the following code:
 
 ```csharp
-MediaFileSystem media = FileSystemProviderManager.Current.GetFileSystemProvider<MediaFileSystem>();
+IMediaFileSystem media = Current.MediaFileSystem;
+```
+or via dependency injection in the constructor for your custom class or controller:
+```csharp
+ public class ImagesController : UmbracoAuthorizedApiController
+    {
+        private readonly IMediaFileSystem _mediaFileSystem;
+
+
+        public ImagesController(IMediaFileSystem mediaFileSystem)
+        {
+            _mediaFileSystem = mediaFileSystem;
+        }
 ```
 
 This will be enough in most cases and is the way Umbraco will access the file system provider. The wrapper class implements the same interface `IFileSystem` as any custom providers should do, so you will be able to call the same methods.
 
-If your custom file system provider has some extra logic that you need to access, you can of course get a reference to the actual instance of IFileSystem. This can be archived through the following code:
+## Setting the PhysicalFileSystem to be outside of the application root
```

Comment:
Is this heading a heading that includes `## MediaPath Scheme` and the rest of the `##` headings in this article? 🤔

Reply:
ahh sorry yes should be removed as i incorporated that bit further up... updated